frontend/src/metabase/ui/components/overlays/Popover/Popover.styled.tsx

```diff
@@ -22,7 +22,8 @@ export const getPopoverOverrides = (): MantineThemeOverride["components"] => ({
       onMouseDownCapture: (event: SyntheticEvent) => {
         // prevent nested popovers from closing each other
         // see useClickOutside in @mantine/hooks for the reference
-        event.nativeEvent.stopImmediatePropagation();
+        const target = event.target as HTMLElement;
+        target.setAttribute("data-ignore-outside-clicks", "true");
```

Comment:
The way event bubbling works in React 17 is now more correct. Previously listeners mounted on `document` would not be affected by `stopImmediatePropagation`, but now they are. Mantine provides this data-attribute for ignoring outside clicks, so we can leverage this for the same effect.